### PR TITLE
TODO #6

### LINKED ISSUE #6: Page: To-do
This task is to create the page (container of components) for To-do. It should contain the following components:

    <todo-header>
    <todo-create-button>
    <todo-toc>
    <todo-list>
    <todo-search>
    <todo-task>
    <todo-new>
    <todo-edit>
    <todo-user-list>
    <todo-comment-history>
    <todo-comment>

image
image

### PR DESCRIPTION
**Pull request by chopcoding**
This task is to create the page (container of components) for To-do. It should contain the following components:

```
<todo-header>
<todo-create-button>
<todo-toc>
<todo-list>
<todo-search>
<todo-task>
<todo-new>
<todo-edit>
<todo-user-list>
<todo-comment-history>
<todo-comment>
```

### CODE REVIEW

File: front/src/components/todo/todo-create-button.vue [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<template>
+  <q-btn @click="create()" class="q-mt-xl text-black float-right" style="background: #F2F2F2" rounded padding='8px 40px' unelevated  no-caps label='Create To Do'/>
+</template>
+<script>
+export default {
+  methods: {
+    create(){
+      this.$root.$emit('CREATE NEW')
```
Comment (onexdata) [inline front/src/components/todo/todo-create-button.vue]: Why are you emitting on $root?

File: front/src/layouts/Todo.vue [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          {
+              completed: false,
+              title: 'Create a Todo List',
+              date: '4:12pm, 12th December to 12:00pm, 14th Decemeber',
+              task: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus et imperdiet augue, nec tincidunt dolor. Praesent ut faucibus magna. Pellentesque ultricies odio eu massa accumsan element quis dolor aliquam pellentesque. Mauris eget.'
+          },
+      ]
+    }
+  },
+  created(){
```
Comment (onexdata) [inline front/src/layouts/Todo.vue]: Setting up root listeners is less than ideal.  Instead, on each component, you can create @eventName listeners.

Also, adding spaces to event names is a bad idea.

Also, not using scoped events will certainly cause conflicts in apps.  For example, a root event for a todo page called "create new" will certainly conflict with "create new" in any other page where you would create something new, as there is no indication that this belongs to the todo page.  Something like "todo.create" would be much more sustainable. However, emmiting on the root bus should always be avoided unless you need to, and in this case you don't even need to.

**Review by onexdata — CHANGES_REQUESTED** [review]
You've set up root listeners and events. This will conflict with the rest of the application.

Please review the "props down, events up" pattern of app building to get a better understanding on how components should be built and how there events should work.

Please see this article for an overview...
https://medium.com/quick-code/vue-communication-patterns-an-intro-to-props-down-and-events-up-pattern-d53340d2c94

**Comment by chopcoding** [comment]
Hi,
Thanks for the feedback, I will make those changes ASAP

**Comment by chopcoding** [comment]
@onexdata 

I have made the necessary changes

**Pull request merged**